engines/scc_suma/app/controllers/scc_suma/scc_suma_controller.rb

```diff
@@ -0,0 +1,84 @@
+require 'json'
+
+module SccSuma
+  REPOSITORY_URL = 'https://scc.suse.com/suma/'.freeze
+  CACHED_PRODUCT_TREE_JSON = '/usr/share/rmt/public/suma/product_tree.json'.freeze
+
+
+  class SccSumaController < ::ApplicationController
+    before_action :is_valid?, only: %w[unscoped_products]
+
+    def unscoped_products
+      update_cache unless cache_is_valid?
+
+      unscoped_products_json = File.read(@unscoped_products_path)
+      render status: :ok, json: { result: JSON.parse(unscoped_products_json) }
+    end
+
+    def list
+      render status: :ok, json: { result: [] }
+    end
+
+    def product_tree
+      render status: :ok, json: { result: product_tree_json }
+    end
+
+    protected
+
+    def scc_client
+      @scc_api_client = SUSE::Connect::Api.new(
+        Settings.scc.username,
+        Settings.scc.password
+      )
+    end
+
+    def is_valid?
+      verification_provider = InstanceVerification.provider.new(
+        logger,
+        request,
+        params.permit(:identifier, :version, :arch, :release_type).to_h,
+        params[:metadata]
+      )
+      raise 'Unspecified error' unless verification_provider.instance_valid?
+    end
+
+    def product_tree_json
+      product_tree_file_path = CACHED_PRODUCT_TREE_JSON
+      unless File.exist?(product_tree_file_path)
+        product_tree_file_path.nil?
+        download_file_from_scc
+        product_tree_file_path = @product_tree_file.local_path
```

Comment:
Why not simply call `update_cache`? I would not expect that we get any performance improvement from the extra implementation of a download routine for the file.

Reply:
Currently, `update_cache` method updates the `unscoped` products info, not the `product_tree`

If you mean `just fetch it every time instead of checking` I thought if the file already existed, one request less was better than doing (in that case) an extra/no needed request

AFAIU, the file is likely to be present, as it gets downloaded when mirroring all enabled repositories, I added the download call in case of mirroring failing or something odd with that file